### PR TITLE
Don't run CI on draft PRs

### PR DESCRIPTION
**Pull request by sd109**

### CODE REVIEW

**Comment by sd109** [comment]
Looks like this triggered CI twice because I originally opened it as a draft with a review request then changed it from draft to ready for review.

**Comment by sd109** [comment]
Would we rather just have CI triggered on 'ready for review' and not 'review requested' so that it doesn't get triggered twice?

File: .github/workflows/demo-deploy.yml [hunk 1]
```diff
@@ -43,6 +43,12 @@ jobs:
           AWS_SECRET_ACCESS_KEY: ${{ secrets.ARCUS_S3_SECRET_KEY }}
 
       - name: Destroy Azimuth
+        # NOTE: This step won't teardown the demo deployment if the CI fails,
+        # which allows us to then access the failed partial deployment to debug.
+        # Assuming you have Arcus cloud access, the seed node for a given CI run
+        # can be accessed by activating the CI environment locally and setting 
+        # the AZIMUTH_ENVIRONMENT env var manually to ci-<github-workflow-run-id> 
+        # then running the seed-ssh script as normal.
```
Comment (mkjpryor) [inline .github/workflows/demo-deploy.yml:51]: Actually, you can specify it as an extra argument to `source ./bin/activate`, which is what the `ci-exec` script does, e.g.:

```
source ./bin/activate ci ci-<run-id>
```

Comment (mkjpryor) [reply .github/workflows/demo-deploy.yml:51]: And, actually, documented here in the GitLab context: https://stackhpc.github.io/azimuth-config/deployment/automation/#per-branch-dynamic-review-environments

**Review by mkjpryor — CHANGES_REQUESTED** [review]

File: .github/workflows/demo-deploy.yml (hunk 1 above)
Comment (sd109) [reply .github/workflows/demo-deploy.yml]: Thanks, I'd forgotten about this

**Review by mkjpryor — DISMISSED** [review]
LGTM

**Review by mkjpryor — APPROVED** [review]
LGTM

**Pull request merged**